src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -212,6 +213,8 @@ internal enum Label
             GroupClauseLambda,                // tied to parent
             QueryContinuation,                // tied to parent
 
+            ForEachComponentStatement,
```

Comment:
Thinking about it more, I think we actually don't want a separate label for ```ForEachComponentStatement```. If we have one than regular foreach won't ever be matched with component-foreach. For example, 

If an edit is made like so:

```C#
foreach (var (a, b) in F()) {...}
```

to 

```C#
foreach (var pair in F()) {...}
```

These won't match at all. That's not desirable, especially if you have an active statement inside the block. 
We should use ForEachStatement for both component and regular foreach and handle transitions between these two forms.